system/mp/mpcomm.cpp

```diff
@@ -2531,28 +2526,54 @@ int CMPConnectThread::run()
     LOG(MCdebugInfo, "MP: Connect Thread Starting - accept loop");
 #endif
     Owned<IException> exception;
+
+    CConnectSelectHandler connectSelectHandler(*this);
     while (running)
     {
         Owned<ISocket> sock;
-        SocketEndpoint peerEp;
+        SocketEndpoint peerEP;
         try
         {
-            sock.setown(listensock->accept(true, &peerEp));
+            sock.setown(listensock->accept(true, &peerEP));
         }
         catch (IException *e)
         {
             exception.setown(e);
         }
         if (sock)
         {
-            if (threadPool)
+#if defined(_USE_OPENSSL)
+            if (parent->useTLS)
             {
-                // if enabled, handle initial connection protocol on a separate thread
-                // if any block for more than a short period (5 seconds), then they are handed off to slowClientProcessor
-                threadPool->start(sock);
+                Owned<ISecureSocket> ssock = secureContextServer->createSecureSocket(sock.getClear());
+                int tlsTraceLevel = SSLogMin;
+                if (parent->mpTraceLevel >= MPVerboseMsgThreshold)
+                    tlsTraceLevel = SSLogMax;
+                int status = ssock->secure_accept(tlsTraceLevel);
+                if (status < 0)
+                {
+                    ssock->close();
+                    PROGLOG("MP Connect Thread: failed to accept secure connection");
+                    continue;
+                }
+                sock.setown(ssock.getClear());
             }
-            else
-                handleAcceptedSocket(sock.getClear(), CONFIRM_TIMEOUT, true);
+#endif // OPENSSL
+
+#ifdef _FULLTRACE
+            StringBuffer s;
+            SocketEndpoint ep1;
+            sock->getPeerEndpoint(ep1);
+            PROGLOG("MP: Connect Thread: socket accepted from %s",ep1.getEndpointHostText(s).str());
+#endif
+            sock->set_keep_alive(true);
```

Comment:
a new call.  Worth a comment why?

Reply:
it's not new really. It's moved and earlier.
It's now upfront/early, here after the accept, before adding to the new connection select handler.
It used to be in handleAcceptedSocket(), which was also logically after the accept() (though delayed until threadpool thread started)